### ISSUE TITLE
[BUG]: SageMaker + S3 artifact store fails trying to create a new bucket

### ISSUE THREAD

**Issue by danguitavinas** [zenml-io/zenml#767]
### System Information

Zenml == 0.10.0

### What happened?

Zenml is trying to create a s3 bucket and fails due to incorrect regex in its name.

### Reproduction steps

1. Create a SageMaker pipeline.
2. Create a s3 artifact store.
3. Run the pipeline

### Relevant log output

```shell
Creating run for pipeline: mnist_pipeline
Cache enabled for pipeline mnist_pipeline
Using stack sagemaker_stack to run pipeline mnist_pipeline...
Step importer has started.
Using cached version of importer.
Step importer has finished in 0.045s.
Step trainer has started.
INFO:botocore.credentials:Found credentials in shared credentials file: ~/.aws/credentials
┌───────────────────── Traceback (most recent call last) ─────────────────────┐
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\s3fs\ │
│ core.py:752 in _mkdir                                                       │
│                                                                             │
│    749 │   │   │   │   │   params["CreateBucketConfiguration"] = {          │
│    750 │   │   │   │   │   │   "LocationConstraint": region_name            │
│    751 │   │   │   │   │   }                                                │
│ >  752 │   │   │   │   await self._call_s3("create_bucket", **params)       │
│    753 │   │   │   │   self.invalidate_cache("")                            │
│    754 │   │   │   │   self.invalidate_cache(bucket)                        │
│    755 │   │   │   except ClientError as e:                                 │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\s3fs\ │
│ core.py:302 in _call_s3                                                     │
│                                                                             │
│    299 │   │   │   except Exception as e:                                   │
│    300 │   │   │   │   err = e                                              │
│    301 │   │   err = translate_boto_error(err)                              │
│ >  302 │   │   raise err                                                    │
│    303 │                                                                    │
│    304 │   call_s3 = sync_wrapper(_call_s3)                                 │
│    305                                                                      │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\s3fs\ │
│ core.py:282 in _call_s3                                                     │
│                                                                             │
│    279 │   │   additional_kwargs = self._get_s3_method_kwargs(method, *akwa │
│    280 │   │   for i in range(self.retries):                                │
│    281 │   │   │   try:                                                     │
│ >  282 │   │   │   │   out = await method(**additional_kwargs)              │
│    283 │   │   │   │   return out                                           │
│    284 │   │   │   except S3_RETRYABLE_ERRORS as e:                         │
│    285 │   │   │   │   logger.debug("Retryable error: %s", e)               │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\aiobo │
│ tocore\client.py:198 in _make_api_call                                      │
│                                                                             │
│   195 │   │   │   'has_streaming_input': operation_model.has_streaming_inpu │
│   196 │   │   │   'auth_type': operation_model.auth_type,                   │
│   197 │   │   }                                                             │
│ > 198 │   │   request_dict = await self._convert_to_request_dict(           │
│   199 │   │   │   api_params, operation_model, context=request_context)     │
│   200 │   │   resolve_checksum_context(request_dict, operation_model, api_p │
│   201                                                                       │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\aiobo │
│ tocore\client.py:246 in _convert_to_request_dict                            │
│                                                                             │
│   243 │                                                                     │
│   244 │   async def _convert_to_request_dict(self, api_params, operation_mo │
│   245 │   │   │   │   │   │   │   │   │      context=None):                 │
│ > 246 │   │   api_params = await self._emit_api_params(                     │
│   247 │   │   │   api_params, operation_model, context)                     │
│   248 │   │   request_dict = self._serializer.serialize_to_request(         │
│   249 │   │   │   api_params, operation_model)                              │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\aiobo │
│ tocore\client.py:275 in _emit_api_params                                    │
│                                                                             │
│   272 │   │                                                                 │
│   273 │   │   event_name = (                                                │
│   274 │   │   │   'before-parameter-build.{service_id}.{operation_name}')   │
│ > 275 │   │   await self.meta.events.emit(                                  │
│   276 │   │   │   event_name.format(                                        │
│   277 │   │   │   │   service_id=service_id,                                │
│   278 │   │   │   │   operation_name=operation_name),                       │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\aiobo │
│ tocore\hooks.py:29 in _emit                                                 │
│                                                                             │
│   26 │   │   │   if asyncio.iscoroutinefunction(handler):                   │
│   27 │   │   │   │   response = await handler(**kwargs)                     │
│   28 │   │   │   else:                                                      │
│ > 29 │   │   │   │   response = handler(**kwargs)                           │
│   30 │   │   │                                                              │
│   31 │   │   │   responses.append((handler, response))                      │
│   32 │   │   │   if stop_on_response and response is not None:              │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\botoc │
│ ore\handlers.py:243 in validate_bucket_name                                 │
│                                                                             │
│    240 │   │   │   'Invalid bucket name "%s": Bucket name must match '      │
│    241 │   │   │   'the regex "%s" or be an ARN matching the regex "%s"' %  │
│    242 │   │   │   │   bucket, VALID_BUCKET.pattern, VALID_S3_ARN.pattern)) │
│ >  243 │   │   raise ParamValidationError(report=error_msg)                 │
│    244                                                                      │
│    245                                                                      │
│    246 def sse_md5(params, **kwargs):                                       │
└─────────────────────────────────────────────────────────────────────────────┘
ParamValidationError: Parameter validation failed:
Invalid bucket name "zenml-training\trainer\.system\executor_execution\24": 
Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$" or be an ARN 
matching the regex 
"^arn:(aws).*:(s3|s3-object-lambda):[a-z\-0-9]*:[0-9]{12}:accesspoint[/:][a-zA-
Z0-9\-.]{1,63}$|^arn:(aws).*:s3-outposts:[a-z\-0-9]+:[0-9]{12}:outpost[/:][a-zA
-Z0-9\-]{1,63}[/:]accesspoint[/:][a-zA-Z0-9\-]{1,63}$"

During handling of the above exception, another exception occurred:

┌───────────────────── Traceback (most recent call last) ─────────────────────┐
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\run-sagemaker.py:87 in       │
│ <module>                                                                    │
│                                                                             │
│   84 │   │   trainer=trainer(),                                             │
│   85 │   │   evaluator=evaluator(),                                         │
│   86 │   )                                                                  │
│ > 87 │   pipeline.run()                                                     │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\zenml │
│ \pipelines\base_pipeline.py:489 in run                                      │
│                                                                             │
│   486 │   │   self._reset_step_flags()                                      │
│   487 │   │   self.validate_stack(stack)                                    │
│   488 │   │                                                                 │
│ > 489 │   │   return stack.deploy_pipeline(                                 │
│   490 │   │   │   self, runtime_configuration=runtime_configuration         │
│   491 │   │   )                                                             │
│   492                                                                       │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\zenml │
│ \stack\stack.py:595 in deploy_pipeline                                      │
│                                                                             │
│   592 │   │   │   pipeline=pipeline, runtime_configuration=runtime_configur │
│   593 │   │   )                                                             │
│   594 │   │                                                                 │
│ > 595 │   │   return_value = self.orchestrator.run(                         │
│   596 │   │   │   pipeline, stack=self, runtime_configuration=runtime_confi │
│   597 │   │   )                                                             │
│   598                                                                       │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\zenml │
│ \orchestrators\base_orchestrator.py:212 in run                              │
│                                                                             │
│   209 │   │   │   pipeline=pipeline, pb2_pipeline=pb2_pipeline              │
│   210 │   │   )                                                             │
│   211 │   │                                                                 │
│ > 212 │   │   result = self.prepare_or_run_pipeline(                        │
│   213 │   │   │   sorted_steps=sorted_steps,                                │
│   214 │   │   │   pipeline=pipeline,                                        │
│   215 │   │   │   pb2_pipeline=pb2_pipeline,                                │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\zenml │
│ \orchestrators\local\local_orchestrator.py:68 in prepare_or_run_pipeline    │
│                                                                             │
│   65 │   │                                                                  │
│   66 │   │   # Run each step                                                │
│   67 │   │   for step in sorted_steps:                                      │
│ > 68 │   │   │   self.run_step(                                             │
│   69 │   │   │   │   step=step,                                             │
│   70 │   │   │   │   run_name=runtime_configuration.run_name,               │
│   71 │   │   │   │   pb2_pipeline=pb2_pipeline,                             │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\zenml │
│ \orchestrators\base_orchestrator.py:316 in run_step                         │
│                                                                             │
│   313 │   │   # This is where the step actually gets executed using the     │
│   314 │   │   # component_launcher                                          │
│   315 │   │   repo.active_stack.prepare_step_run()                          │
│ > 316 │   │   execution_info = self._execute_step(component_launcher)       │
│   317 │   │   repo.active_stack.cleanup_step_run()                          │
│   318 │   │                                                                 │
│   319 │   │   return execution_info                                         │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\zenml │
│ \orchestrators\base_orchestrator.py:340 in _execute_step                    │
│                                                                             │
│   337 │   │   start_time = time.time()                                      │
│   338 │   │   logger.info(f"Step `{pipeline_step_name}` has started.")      │
│   339 │   │   try:                                                          │
│ > 340 │   │   │   execution_info = tfx_launcher.launch()                    │
│   341 │   │   │   if execution_info and get_cache_status(execution_info):   │
│   342 │   │   │   │   logger.info(f"Using cached version of `{pipeline_step │
│   343 │   │   except RuntimeError as e:                                     │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\tfx\o │
│ rchestration\portable\launcher.py:528 in launch                             │
│                                                                             │
│   525 │   │   │   │   │   │   │   │   │   │      self._pipeline_runtime_spe │
│   526 │                                                                     │
│   527 │   # Runs as a normal node.                                          │
│ > 528 │   execution_preparation_result = self._prepare_execution()          │
│   529 │   (execution_info, contexts,                                        │
│   530 │    is_execution_needed) = (execution_preparation_result.execution_i │
│   531 │   │   │   │   │   │   │    execution_preparation_result.contexts,   │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\tfx\o │
│ rchestration\portable\launcher.py:388 in _prepare_execution                 │
│                                                                             │
│   385 │   │   │     output_dict=output_artifacts,                           │
│   386 │   │   │     exec_properties=exec_properties,                        │
│   387 │   │   │     execution_output_uri=(                                  │
│ > 388 │   │   │   │     self._output_resolver.get_executor_output_uri(execu │
│   389 │   │   │     stateful_working_dir=(                                  │
│   390 │   │   │   │     self._output_resolver.get_stateful_working_director │
│   391 │   │   │     tmp_dir=self._output_resolver.make_tmp_dir(execution.id │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\tfx\o │
│ rchestration\portable\outputs_utils.py:172 in get_executor_output_uri       │
│                                                                             │
│   169 │   """Generates executor output uri given execution_id."""           │
│   170 │   execution_dir = os.path.join(self._node_dir, _SYSTEM, _EXECUTOR_E │
│   171 │   │   │   │   │   │   │   │    str(execution_id))                   │
│ > 172 │   fileio.makedirs(execution_dir)                                    │
│   173 │   return os.path.join(execution_dir, _EXECUTOR_OUTPUT_FILE)         │
│   174                                                                       │
│   175   def get_driver_output_uri(self) -> str:                             │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\tfx\d │
│ sl\io\fileio.py:80 in makedirs                                              │
│                                                                             │
│    77                                                                       │
│    78 def makedirs(path: PathType) -> None:                                 │
│    79   """Make a directory at the given path, recursively creating parents │
│ >  80   _get_filesystem(path).makedirs(path)                                │
│    81                                                                       │
│    82                                                                       │
│    83 def mkdir(path: PathType) -> None:                                    │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\zenml │
│ \integrations\s3\artifact_stores\s3_artifact_store.py:275 in makedirs       │
│                                                                             │
│   272 │   │   Args:                                                         │
│   273 │   │   │   path: The path to create.                                 │
│   274 │   │   """                                                           │
│ > 275 │   │   self.filesystem.makedirs(path=path, exist_ok=True)            │
│   276 │                                                                     │
│   277 │   def mkdir(self, path: PathType) -> None:                          │
│   278 │   │   """Create a directory at the given path.                      │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\fsspe │
│ c\asyn.py:85 in wrapper                                                     │
│                                                                             │
│    82 │   @functools.wraps(func)                                            │
│    83 │   def wrapper(*args, **kwargs):                                     │
│    84 │   │   self = obj or args[0]                                         │
│ >  85 │   │   return sync(self.loop, func, *args, **kwargs)                 │
│    86 │                                                                     │
│    87 │   return wrapper                                                    │
│    88                                                                       │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\fsspe │
│ c\asyn.py:65 in sync                                                        │
│                                                                             │
│    62 │   │   # suppress asyncio.TimeoutError, raise FSTimeoutError         │
│    63 │   │   raise FSTimeoutError from return_result                       │
│    64 │   elif isinstance(return_result, BaseException):                    │
│ >  65 │   │   raise return_result                                           │
│    66 │   else:                                                             │
│    67 │   │   return return_result                                          │
│    68                                                                       │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\fsspe │
│ c\asyn.py:25 in _runner                                                     │
│                                                                             │
│    22 │   if timeout is not None:                                           │
│    23 │   │   coro = asyncio.wait_for(coro, timeout=timeout)                │
│    24 │   try:                                                              │
│ >  25 │   │   result[0] = await coro                                        │
│    26 │   except Exception as ex:                                           │
│    27 │   │   result[0] = ex                                                │
│    28 │   finally:                                                          │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\s3fs\ │
│ core.py:767 in _makedirs                                                    │
│                                                                             │
│    764 │                                                                    │
│    765 │   async def _makedirs(self, path, exist_ok=False):                 │
│    766 │   │   try:                                                         │
│ >  767 │   │   │   await self._mkdir(path, create_parents=True)             │
│    768 │   │   except FileExistsError:                                      │
│    769 │   │   │   if exist_ok:                                             │
│    770 │   │   │   │   pass                                                 │
│                                                                             │
│ C:\Users\i25262\PycharmProjects\ZenML-Kubeflow\venv\lib\site-packages\s3fs\ │
│ core.py:758 in _mkdir                                                       │
│                                                                             │
│    755 │   │   │   except ClientError as e:                                 │
│    756 │   │   │   │   raise translate_boto_error(e)                        │
│    757 │   │   │   except ParamValidationError as e:                        │
│ >  758 │   │   │   │   raise ValueError("Bucket create failed %r: %s" % (bu │
│    759 │   │   else:                                                        │
│    760 │   │   │   # raises if bucket doesn't exist and doesn't get create  │
│    761 │   │   │   await self._ls(bucket)                                   │
└─────────────────────────────────────────────────────────────────────────────┘
ValueError: Bucket create failed 
'zenml-training\\trainer\\.system\\executor_execution\\24': Parameter 
validation failed:
Invalid bucket name "zenml-training\trainer\.system\executor_execution\24": 
Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$" or be an ARN 
matching the regex 
"^arn:(aws).*:(s3|s3-object-lambda):[a-z\-0-9]*:[0-9]{12}:accesspoint[/:][a-zA-
Z0-9\-.]{1,63}$|^arn:(aws).*:s3-outposts:[a-z\-0-9]+:[0-9]{12}:outpost[/:][a-zA
-Z0-9\-]{1,63}[/:]accesspoint[/:][a-zA-Z0-9\-]{1,63}$"
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by schustmi** [comment]
Hi @danguitavinas,

I'm guessing from the stack trace that you're running on windows with the local orchestrator? If that's the case, my guess is that this issue should be fixed by #735.

If you're interested in trying this, you could install ZenML from that branch using the command `pip install git+https://github.com/zenml-io/zenml.git@bugfix/windows-source-utils`

**Comment by danguitavinas** [comment]
Thank you so much, that worked! Im closing the issue!
